### ISSUE TITLE
SelectOneMenu: client side function selectValue() doesn't work before menu was opened

### ISSUE THREAD

**Issue by cnsgithub** [primefaces/primefaces#7393]
**Describe the defect**
When calling the SelectOneMenu widget function [selectValue()](https://primefaces.github.io/primefaces/jsdocs/classes/src_primefaces.primefaces.widget.selectonemenu-1.html#selectvalue) before the menu was opened at least once, the function call fails.

````
Uncaught TypeError: Cannot read property 'eq' of undefined
    at c.selectValue (components.js.xhtml?…mefaces&v=10.0.0:20)
    at <anonymous>:1:45
selectValue	@	components.js.xhtml?…mefaces&v=10.0.0:20
(anonymous)	@	VM812:1
````
It seems like ``this.items`` in [forms.selectonemenu.js:1114](https://github.com/primefaces/primefaces/blob/10.0.0/src/main/resources/META-INF/resources/primefaces/forms/forms.selectonemenu.js#L1114) is undefined in this case.

After opening the menu, everything works fine.

**Reproducer**
Showcase for SelectOneMenu https://www.primefaces.org/showcase/ui/input/oneMenu.xhtml

**Environment:**
 - PF Version: 10.0.0
 - JSF + version: MyFaces 2.3.6
 - Affected browsers: ALL

**To Reproduce**
1. Open the SelectOneMenu showcase https://www.primefaces.org/showcase/ui/input/oneMenu.xhtml
2. Execute the ``selectValue()`` function in the console for one of the menus, e.g. 
````js
PrimeFaces.getWidgetById('j_idt302:option').selectValue('Option1')
````
3.  If you open the menu, e.g. ``PrimeFaces.getWidgetById('j_idt302:option').show()`` and then execute 2. again, it works.

**Expected behavior**
``selectValue()`` should be able to select a value even before the menu was opened.

**Comment by melloware** [comment]
I think this was introduced with this issue: https://github.com/primefaces/primefaces/issues/6418

Looks like if `selectValue` is called an the items are empty it has to create the items.

**Comment by melloware** [comment]
This MonkeyPatch will fix you up..

```javascript
if (PrimeFaces.widget.SelectOneMenu) {

    PrimeFaces.widget.SelectOneMenu.prototype.callHandleMethod = function(handleMethod, event) {
        var $this = this;

        if (this.cfg.dynamic && !this.isDynamicLoaded) {
            this.dynamicPanelLoad();

            var interval = setInterval(function() {
                if ($this.isDynamicLoaded) {
                    if (handleMethod) {
                        handleMethod.call($this, event);
                    }

                    clearInterval(interval);
                }
            }, 10);
        } else {
            this.renderPanelContentFromHiddenSelect(true);

            if (handleMethod) {
                handleMethod.call(this, event);
            }
        }
    };

    PrimeFaces.widget.SelectOneMenu.prototype.selectValue = function(value) {
        if (!this.items || this.items.length === 0) {
            this.callHandleMethod(null, null);
        }

        var option = this.options.filter('[value="' + $.escapeSelector(value) + '"]');

        this.selectItem(this.items.eq(option.index()), true);
    };
}
```

**Comment by cnsgithub** [comment]
Thanks for the quick fix. I just tried it with a build that includes 5a5be48 and the selectValue() function now works as expected.

**Comment by mn-qualitasag** [comment]
Could that also be included in a 10.0.x patch version?

**Comment by melloware** [comment]
I marked for Elite but its up to @sqores to implement.
